### PR TITLE
[Build Profile] Optimising the relase build profile

### PR DESCRIPTION
**Pull request by Markuze**
### Description
Optimize release build config
 
### Test Plan

Ran forge with different release profile configs: 
"text": "all up : 8567 TPS, 3451 ms latency, 5000 ms p99 latency,no expired txns

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2575)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -148,8 +148,11 @@ default-members = [
 ]
 
 [profile.release]
+opt-level = 3
 debug = true
-overflow-checks = true
+overflow-checks = false
```
Comment (zekun000) [inline Cargo.toml:153]: @msmouse ^^

**Comment by zekun000** [comment]
I wonder how much this slows down the build time

File: Cargo.toml (hunk 1 above)
Comment (Markuze) [reply Cargo.toml:153]: The cost of this flag is about 7% in performance on forge.
I can add a separate [profile.performance] which will differ from release by the value of overflow-checks.

Comment (Markuze) [reply Cargo.toml:153]: [source](https://doc.rust-lang.org/cargo/reference/profiles.html#dev)
The overflow-checks setting controls the [-C overflow-checks flag](https://doc.rust-lang.org/rustc/codegen-options/index.html#overflow-checks) which controls the behavior of [runtime integer overflow](https://doc.rust-lang.org/reference/expressions/operator-expr.html#overflow). When overflow-checks are enabled, a panic will occur on overflow.

File: Cargo.toml [hunk 2]
```diff
@@ -148,8 +148,11 @@ default-members = [
...
+lto = true
+codegen-units = 1
```
Comment (sitalkedia) [inline Cargo.toml:155]: This will slow down the build time significantly I believe?

**Comment by sitalkedia** [comment]
@Markuze - Out of curiousity, do you know which of the above change helped boost the TPS the most?

File: Cargo.toml (hunk 2 above)
Comment (Markuze) [reply Cargo.toml:155]: It does. If this impacts dev tme, I can just add another profile, that we can use discriminately.

**Comment by Markuze** [comment]
overflow check adds a negative 700 TPS. Codegen units adds 500TPS. 
 Its not clear to me if there is a difference between opt-level 1 2 and 3.
I assume that the biggest impact is from LTO.

File: Cargo.toml (hunk 2 above)
Comment (zekun000) [reply Cargo.toml:155]: let's make a different profile, release build is already slow enough

Comment (Markuze) [reply Cargo.toml:160]: all up : 8501 TPS, 3478 ms latency, 6000 ms p99 latency,no expired txns.
With 28m build time. Now gated behind `cargo build --profile performance`

**Review by zekun000 — APPROVED** [review]

**Pull request merged**